pulsar-client-api/src/main/java/org/apache/pulsar/client/internal/ReflectionUtils.java

```diff
@@ -47,23 +62,17 @@
     }
 
     @SuppressWarnings("unchecked")
-    static <T> Class<T> newClassInstance(String className) {
-        try {
+    public static <T> Class<T> newClassInstance(String className) {
+        return (Class<T>) loadedClasses.computeIfAbsent(className, name -> {
```

Comment:
why do we need this cache ?
the JVM is very good in dealing with Class.forName.